src/test/scala/RepoTest.scala

```diff
@@ -0,0 +1,74 @@
+import org.scalatest._
+import Matchers.{be, _}
+import cats.instances.future._
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.language.higherKinds
+import scala.concurrent.{Await, Future}
+import scala.concurrent.duration._
+
+
+class RepoTest extends FlatSpec {
+  val idUserRepository = new IdUserRep()
+  val idDeviceRepository =  new IdDeviceRep()
+
+  val users = List("Bob", "Jon", "Ava")
+  val devices = List("device1", "device2", "device3")
+
+  val userService = new UserService(idUserRepository)
+  val deviceService = new IotDeviceService(idDeviceRepository, idUserRepository)
+
+  it should s"Created Users $users" in {
+    users.map(user => userService.registerUser(user))
+    users.map(userName => userService.getByUsername(userName).fold("")(user => user.username) shouldEqual userName)
+    users.zipWithIndex.foreach{ case (userName, i) => userService.getById(i).fold("")(user => user.username) shouldEqual userName }
+  }
+
+  it should s"Created Users with exist userName" in {
+    users.map(user => userService.registerUser(user) should be ('left))
+  }
+
+  it should s"Created Devices $devices" in {
+    users.zipWithIndex.foreach{ case (_, userId) => {
+      devices.map(deviceSn => deviceService.registerDevice(userId, s"$userId-$deviceSn") should be ('right))
+    }
+    }
+  }
+
+  it should s"Created Devices with exist serial number" in {
+    users.zipWithIndex.foreach{ case (_, userId) => {
+      devices.map(deviceSn => deviceService.registerDevice(userId, s"$userId-$deviceSn") should be ('left))
+    }
+    }
+  }
+
+  def await[T](f: Future[T]) = Await.result(f, 2.second)
```

Comment:
you can use AsyncFlatSpec and get rid of await, more info at http://www.scalatest.org/user_guide/async_testing